enterprise/admin/views.py

```diff
@@ -205,11 +205,13 @@ def _handle_singular(cls, enterprise_customer, manage_learners_form):
         form_field_value = manage_learners_form.cleaned_data[ManageLearnersForm.Fields.EMAIL_OR_USERNAME]
         email = email_or_username__to__email(form_field_value)
         try:
-            validate_email_to_link(email, form_field_value, ValidationMessages.INVALID_EMAIL_OR_USERNAME)
+            validate_email_to_link(email, form_field_value, ValidationMessages.INVALID_EMAIL_OR_USERNAME, True)
         except ValidationError as exc:
             manage_learners_form.add_error(ManageLearnersForm.Fields.EMAIL_OR_USERNAME, exc)
         else:
-            EnterpriseCustomerUser.objects.link_user(enterprise_customer, email)
+            if not (EnterpriseCustomerUser.objects.is_user_linked(enterprise_customer, email) or
+                    EnterpriseCustomerUser.objects.is_user_pending(enterprise_customer, email)):
+                EnterpriseCustomerUser.objects.link_user(enterprise_customer, email)
```

Comment:
I think it would be better to modify the [link_user](https://github.com/edx/edx-enterprise/blob/master/enterprise/models.py#L305) function to call [get_or_create](https://docs.djangoproject.com/en/dev/ref/models/querysets/#get-or-create) instead of create.

Reply:
will give that a try, didn't know about that method. agreed would be cleaner